node/core/pvf/src/host.rs

```diff
@@ -507,9 +486,8 @@ async fn handle_execute_pvf(
 	awaiting_prepare: &mut AwaitingPrepare,
 	inputs: ExecutePvfInputs,
 ) -> Result<(), Fatal> {
-	let ExecutePvfInputs { pvf_with_params, execution_timeout, params, priority, result_tx } =
-		inputs;
-	let artifact_id = pvf_with_params.as_artifact_id();
+	let ExecutePvfInputs { pvf, execution_timeout, params, priority, result_tx } = inputs;
```

Comment:
Should the `execution_timeout` be part of the `pvf` now?

Reply:
I believe it shouldn't as the execution timeouts are per job, not per artifact.

Reply:
Looks like in `validate_candidate_with_retry` we construct the `pvf` from the timeout
:
```rs
let pvf = PvfExhaustive::from_code(raw_validation_code, executor_params, timeout);
```

but then we still pass along the PVF and timeout:

```rs
validation_result = self.validate_candidate(pvf, timeout, params.encode()).await;
```

Reply:
Exactly. The timeout inside `PvfExhaustive` is the preparation timeout, and the one passed to `validate_candidate()` is the execution timeout. PVF host may or may not have an already prepared artifact by the time of validation request, so it needs both. But the naming is dubious, I think they should be renamed.

Reply:
Oh, I see. Yeah, renaming it to `preparation_timeout` would be better. PVFs can be either prepared or executed so we should distinguish what the timeout is for in its name.